libs/coroutines/src/detail/coroutine_impl.cpp

```diff
@@ -96,6 +96,8 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
                         result_last.first == thread_state_enum::terminated);
                 }
 
+                this->reset_tss();
+
```

Comment:
Shouldn't the rest of `this` be reset somewhere now as well?

Reply:
Ahh, it is - please ignore me.

Reply:
But here is a real question: shouldn't the TSS be reset only if the `state == terminated`?

Reply:
https://github.com/STEllAR-GROUP/hpx/pull/4137/files/db357236eaa7cc7b97c682ac9cb0181ceb4cfa15#diff-ae32f64c0aa2f9db307e2e779cf067e3R95

What about this?

Reply:
Did you come to a consensus on whether TSS should be reset if there's an exception?

Reply:
Yes, it should be reset if there is an exception. I just (force) pushed a fix to remedy this.